### DISCUSSION TITLE
Writing a script that will take a single photo every day at the same time?

### DISCUSSION THREAD

**Discussion by videoword** [gopro/labs#843] [Q&A]
Hi heroes, I just got a GoPro 9 and I'm trying to wrap my head around this stuff. Downloaded the QR generator and been going through the scripting references. Here is what I've come up with.

What I'd like to do in English: Turn on camera at 2:00pm, take a single photo, turn off camera, repeat again at 2pm.

!14:00NmPdP!S!14:00RQ

or should it be...

!14:00NmPdP!S!86400RQ

or something else entirely...

Thanks!

**Comment by dnewman-gpsw** [comment]
Very simple, it just !14:00NmP!S!1R

Sleep until 2pm
Set mode Photo
Take Photo
Repeat

As it is slightly past 2pm the sleep will wake 24 hours later.  If take more than a photo a day, that is a bug.  Let me know.

**Reply by videoword** [reply]
Ok, great! Trying this out today. I understand all the script, except the '1' in the '!1R'. Is that just telling the camera to wait 1 second before repeating from the beginning?

**Reply by dnewman-gpsw** [reply]
Correct.  I add the 1 second pause so loops don't happen too quickly.  This is generally not necessary, so !R would be fine. It is just good practice for other type of scripts.

**Comment by videoword** [comment]
Makes sense. I waited until 2pm today, and then went and checked the camera roll. No photo.

Been wracking my brain about it for a while now and finally figured out that every time I use the QR code to give the gopro the instructions, the internal time resets.

If I set the time and turn the gopro off, it's all good even when I restart. But as soon as I give it the QR code, it turns off and resets the time.

I've tried resetting the time from the camera as well as from my phone app.

Do I just need to run with it and pretend that it is actually 7pm right now?

**Reply by dnewman-gpsw** [reply]
So you are not running the command I sent, you are telling it to set the time on every repeat do not do that.  Turn off "Set date + time automatically".

The command need to be ```!14:00NmP!S!R``` and nothing else.

**Reply by videoword** [reply]
I'm sure you're right and I'm doing it wrong, but here is a screengrab of the code I'm scanning (I did change the time here because I'm trying to test it again today, but I think that's the only thing I changed?):

![Screenshot 2024-04-05 at 2 59 57 PM](https://github.com/gopro/labs/assets/166088235/c6bd0a86-3b06-4ed0-825c-8086ae5ff653)

I tried it again, and it changes the time to '19:21' every time I scan it.

**Reply by dnewman-gpsw** [reply]
That QR code is wrong, it has date and time in it. 

This is the correct one:
![image](https://github.com/gopro/labs/assets/10856084/b00d311c-2d63-48cc-a71e-10dcdd1f1db5)

**Reply by videoword** [reply]
So weird! Yeah, yours looks totally different and it works.
